lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -285,9 +284,11 @@ def request_headers(options)
         headers = {
           'Content-Type' => 'application/x-www-form-urlencoded'
         }
-        headers['CompanyName'] = options[:company_name]
         headers['AccessToken'] = @access_token
         headers['Invoice'] = options[:invoice] if options[:invoice].present?
+        headers['InterfaceVersion'] = options[:interface_version] if options[:interface_version]
+        headers['InterfaceName'] = options[:interface_name] if options[:interface_name]
+        headers['CompanyName'] = options[:company_name] if options[:company_name]
```

Comment:
The conversation we had about the `AccessToken` header applies here too.

Net/HTTP will convert all of these headers to sentence-case (e.g. `Companyname`, `Interfacename`, `Interfaceversion`).

You don't need to change anything with these headers right now. We can just add them this way and wait to hear back from Shift4 about whether they will be able to loosen their requirements about the case of the headers.